core-api/src/main/java/com/optimizely/ab/config/ProjectConfig.java

```diff
@@ -170,6 +173,14 @@ public ProjectConfig(String accountId,
 
         this.attributes = Collections.unmodifiableList(attributes);
         this.audiences = Collections.unmodifiableList(audiences);
+
+        if (typedAudiences != null) {
+            this.typedAudiences = Collections.unmodifiableList(typedAudiences);
+        }
+        else {
+            this.typedAudiences = Collections.emptyList();
```

Comment:
Wouldn't it make life easier if `this.typedAudiences = Collections.unmodifiableList(audiences)` instead?